### PR TITLE
Rich Text Editor

### PR DESCRIPTION
**Pull request by franalderete7**

### CODE REVIEW

File: components/FormField.js [hunk 1]
```diff
@@ -186,6 +187,10 @@ const FormField = ({ data, update, className, label, placeholder, name, type, re
             onChange={value => update(name, value)}
           />
           }
+          {
+            type === 'richText' && 
+            <RichText />
```
Comment (closerearth) [inline components/FormField.js:192]: Nice :) Thanks for adding to this file.

Comment (closerearth) [inline components/FormField.js:192]: Nice :) Thanks for adding to this file.

File: utils/helpers.js [hunk 2]
```diff
@@ -130,6 +130,8 @@ export const getSample = (field) => {
   case 'email':
   case 'phone':
     return '';
+  case 'richText':
+    return JSON.stringify();
```
Comment (closerearth) [inline utils/helpers.js:134]: getSample function is meant to return a sample field value. Here you are returning `undefined` which is probably not what you want.

File: components/RichText.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const RichText = () => {
+  const editor = useMemo(() => withReact(createEditor()), [])
+  const [value, setValue] = useState([
+    {
+      type: 'paragraph',
+      children: [{ text: 'A line of text in a paragraph for testing.' }],
+    },
+  ])
+
+  return (
```
Comment (closerearth) [inline components/RichText.js:18]: Can you implement some rich text functionality so we can actually test... bold, italic, header, quote etc.

Comment (closerearth) [inline components/RichText.js:18]: Can you implement some rich text functionality so we can actually test... bold, italic, header, quote etc.

File: utils/helpers.js (hunk 2 above)
Comment (franalderete7) [reply utils/helpers.js:134]: Oh yes thats true

File: components/RichText.js (hunk 3 above)
Comment (franalderete7) [reply components/RichText.js:18]: Yes :)

**Comment by franalderete7** [comment]
Some rich text functionalities implemented such as bold, italic and underline. Try selecting a piece of text.

**Comment by franalderete7** [comment]
If the component is functional, we can decide how to save it.

**Comment by b-triz** [comment]
@franalderete7 super cool! super good! Congrads.
Lets polish some things:

- Instead of saying 'Text Editor' lets write 'Description' and remove the original 'description box'.
- Lets add font colors to the editor.
- Lets do an image uploader.
good?

**Comment by franalderete7** [comment]
I switched libraries because Slate was highly customizable and the code was way too long and hard to implement. Also, it did not support dragging or uploading images from file explorer (It is on Beta). The one I am using right now is easier to work with and it is more similar to the one in Medium.

**Comment by franalderete7** [comment]
The editor now has styles matching the one in Medium, as well as a complete toolbar supporting image and video uploads :).

File: et --soft HEAD~1 [hunk 4]
```diff
@@ -0,0 +1,82 @@
+[33mad7c110[m[33m ([m[1;36mHEAD -> [m[1;32mtext-editor[m[33m, [m[1;31morigin/text-editor[m[33m)[m correct typo
```
Comment (closerearth) [inline et --soft HEAD~1:1]: 👀

File: components/editor/TextEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  createHorizontalRulePlugin,
+  createPlugins,
+  createDeserializeDocxPlugin,
+  createJuicePlugin,
+} from '@udecode/plate'
+import {
+  createExcalidrawPlugin,
+  ELEMENT_EXCALIDRAW,
+  ExcalidrawElement,
+} from '@udecode/plate-ui-excalidraw'
```
Comment (closerearth) [inline components/editor/TextEditor.tsx:55]: Let's remove these third party stuff.

**Comment by closerearth** [comment]
We need to cut off depencies by a lot in the MR.

This increases 4x the size of the page (check `├ λ /listings/[slug]/edit                  984 B           684 kB`)

Screen Shot 2022-04-26 at 3 29 03 PM

Let's only have what's really required for editing here, and see if we can reduce.

**Comment by closerearth** [comment]
Why do you need setting .babelrc ? it reduces performance optimizations done by Nextjs.

**Comment by closerearth** [comment]
What you are saving is definitely not what you want to be saving:

```
[{"type":"p","__source":{"fileName":"/Users/closer/Code/app.closer.earth/components/editor/config/values/values.tsx","lineNumber":12,"columnNumber":5},"children":[{"text":"Write a description."}]},{"type":"p","__source":{"fileName":"/Users/closer/Code/app.closer.earth/components/editor/config/values/values.tsx","lineNumber":12,"columnNumber":5},"id":1651102338369,"children":[{"text":"dafaefaef","bold":true}],"indent":3}]
```

**Comment by closerearth** [comment]
Can't edit the text.

```next-dev.js?3515:25 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
```

**Comment by b-triz** [comment]
@samueldelesque a handful of functionalities were removed from editor, making it less heavy. I prefer this one to the alternative Fran created. See what's up for you!

**Pull request closed without merging**